### PR TITLE
[DX] Rename namespaces respect node type

### PR DESCRIPTION
**Pull request by dobryy**

### CODE REVIEW

**Comment by dobryy** [comment]
@TomasVotruba I bet you gonna hate me after this PR :D Tried to accomplish it as fast as possible that is why commits are not that well organized, sorry for that.

File: phpstan.neon [hunk 1]
```diff
@@ -393,3 +393,12 @@ parameters:
...
 
         - '#Class with base "(.*?)" name is already used in "_HumbugBox(.*?)"#'
+
+        #tmp
+        -
+            message: '#Class (.*?) has invalid namespace category (.*?)\. Pick one of\: (.*?)#'
+            path: 'rules/generic/src/Rector/Architecture'
+
+        -
+            message: '#Class "Rector\\RectorGenerator\\ValueObject\\RectorRecipe" has invalid namespace category "ValueObject"\. Pick one of\: ""#'
```
Comment (dobryy) [inline phpstan.neon:403]: @TomasVotruba Can you please update PHPStan rule to skip this file?

Comment (dobryy) [reply phpstan.neon:403]: Or... we can leave this ignore rule here...

File: phpstan.neon [hunk 2]
```diff
@@ -393,3 +393,12 @@ parameters:
         - '#Parameter \#1 \$type of method Rector\\NodeCollector\\NodeCollector\\ParsedNodeCollector<TNodeType of PhpParser\\Node\>\:\:getNodesByType\(\) expects class\-string<TNodeType of PhpParser\\Node\>, string given#'
 
         - '#Class with base "(.*?)" name is already used in "_HumbugBox(.*?)"#'
+
+        #tmp
+        -
+            message: '#Class (.*?) has invalid namespace category (.*?)\. Pick one of\: (.*?)#'
+            path: 'rules/generic/src/Rector/Architecture'
```
Comment (dobryy) [inline phpstan.neon:400]: Should I move rules in Architecture folder to the `rules/generic/src/Rector/` and use an appropriate namespace for it or I have to create new namespaces in Architecture folder?

File: rules/sensio/tests/Rector/ClassMethod/TemplateAnnotationToThisRenderRector/Fixture/compact_controller.php.inc [hunk 3]
```diff
@@ -1,6 +1,6 @@
 <?php
 
-namespace Rector\Sensio\Tests\Rector\ClassMethod\TemplateAnnotationToThisRenderRector\Fixture;
+namespace Rector\Sensio\Tests\Rector\FrameworkExtraBundle\TemplateAnnotationToThisRenderRector\Fixture;
```
Comment (dobryy) [inline rules/sensio/tests/Rector/ClassMethod/TemplateAnnotationToThisRenderRector/Fixture/compact_controller.php.inc]: @TomasVotruba I'm not sure why when ClassMethod namespace is used in this fixture the test is failing. I have no clue why :/

Comment (TomasVotruba) [reply rules/sensio/tests/Rector/ClassMethod/TemplateAnnotationToThisRenderRector/Fixture/compact_controller.php.inc]: What's inside the rule that depends on this namespace?

Comment (TomasVotruba) [reply rules/sensio/tests/Rector/ClassMethod/TemplateAnnotationToThisRenderRector/Fixture/compact_controller.php.inc]: What's inside the rule that depends on this namespace?

**Comment by TomasVotruba** [comment]
It's not possible to comment/review very well, as there are too many changes now. Can you squash and merge the work so far?

**Comment by TomasVotruba** [comment]
I see that CI passes, I'm merging now

**Comment by tavy315** [comment]
@TomasVotruba is the change of RemoveNullPropertyInitializationRector correct? Was it intented to be moved to PropertyProperty?

image

**Pull request merged**